### PR TITLE
Streamlining of panel addition.

### PR DESCRIPTION
**Pull request by DJKero**
Streamlines https://github.com/teejusb/fsr/issues/8#issuecomment-817170977
You need to add the sensors to fsr.ino and change the "panels" variable on both server.py and app.js

### CODE REVIEW

**Comment by teejusb** [comment]
Thanks! This is something I had on the backburner and hadn't got around to yet. There are a couple of code changes which I'll add to the files.

File: webui/server/server.py [hunk 1]
```diff
@@ -16,16 +16,29 @@
 logger = logging.getLogger(__name__)
 
 # Edit this to match the serial port name shown in Arduino IDE
-SERIAL_PORT = "/dev/ttyACM0"
+SERIAL_PORT = "COM3"
```
Comment (teejusb) [inline webui/server/server.py:19]: Can we stick with ACM0 for the default?

File: webui/server/server.py [hunk 2]
```diff
@@ -16,16 +16,29 @@
...
 HTTP_PORT = 5000
 
 # Threads for the serial reader and writer.
 read_thread = None
 write_thread = None
 thread_stop_event = threading.Event()
 
-# L, D, U, R
-sensor_numbers = [0, 1, 2, 3]
+def init_list(length, incremental = False):
```
Comment (teejusb) [inline webui/server/server.py:27]: We don't actually need this function.

Below do:

If you want an incremental sequence do:
```
sensor_numbers = range(num_panels)
```

If you want a list of zeroes do:
```
[0] * num_panels
```

File: webui/server/server.py [hunk 3]
```diff
@@ -16,16 +16,29 @@
...
+	if incremental == False:
+		for i in range(length):
+			temp.append(0)
+	else:
+		for i in range(length):
+			temp.append(i)
+	return temp
+
+# Amount of panels.
+panels = 5
```
Comment (teejusb) [inline webui/server/server.py:38]: Rename to num_panels

File: webui/src/App.js [hunk 4]
```diff
@@ -21,21 +21,40 @@ import {
   Link
 } from "react-router-dom";
 
+// Amount of panels.
+let panels = 5;
```
Comment (teejusb) [inline webui/src/App.js:25]: num_panels

Comment (teejusb) [reply webui/src/App.js:25]: Also default to 4

File: webui/server/server.py (hunk 3 above)
Comment (teejusb) [reply webui/server/server.py:38]: also default to 4

File: webui/src/App.js [hunk 5]
```diff
@@ -21,21 +21,40 @@ import {
...
+let panel_indexes = init_array(panels, true);
```
Comment (teejusb) [inline webui/src/App.js:26]: This doesn't need to be global and can be next to the ValueMonitor stuff directly.

File: webui/src/App.js [hunk 6]
```diff
@@ -21,21 +21,40 @@ import {
...
-let kCurValues = [[0, 0, 0, 0]]
+let kCurValues = [init_array(panels)];
 const max_size = 1000;
 let oldest = 0;
 
 var ws;
 const wsCallbacks = {};
 const wsQueue = [];
 
+function init_array(length, incremental = false) {
```
Comment (teejusb) [inline webui/src/App.js:43]: I don't think we need this function either.

for an array of zeroes you can do:
```
new Array(num_panels).fill(0)
```

for a sequential list you can do:
```
[...Array(num_panels).keys()]
```

File: webui/src/App.js (hunk 4 above)
Comment (teejusb) [reply webui/src/App.js:25]: also make const

File: webui/src/App.js (hunk 5 above)
Comment (teejusb) [reply webui/src/App.js:26]: You might not even need this variable if you do

```
{[...Array(num_panels).keys()].map(i => (
   <ValueMonitor index={i}/>)
)}
```

below

File: webui/src/App.js (hunk 6 above)
Comment (DJKero) [reply webui/src/App.js:42]: Thanks! ^^"

**Comment by DJKero** [comment]
> 
> 
> Thanks! This is something I had on the backburner and hadn't got around to yet. There are a couple of code changes which I'll add to the files.

No prob! Just made the changes you requested on the review!

**Comment by teejusb** [comment]
Okay cool I was able to test this and it looks good to me!

**Pull request merged**